src/profile-logic/transforms.js

```diff
@@ -572,6 +580,42 @@ export function mergeFunction(
   });
 }
 
+/**
+ * Drop any samples that contain the given function.
+ */
+export function dropFunction(
+  thread: Thread,
+  funcIndexToDrop: IndexIntoFuncTable
+) {
+  const { stackTable, frameTable, samples } = thread;
+
+  // Go through each stack, and label it as containing the function or not.
+  const stackContainsFunc = new Set();
```

Comment:
I wonder if an array of booleans would be faster here. Would require less hashing.

Reply:
Ah good point, I'll make that change.